server/src/main/java/org/apache/druid/client/coordinator/CoordinatorClientImpl.java

```diff
@@ -71,30 +71,13 @@ public ListenableFuture<Boolean> isHandoffComplete(String dataSource, SegmentDes
   }
 
   @Override
-  public ListenableFuture<DataSegment> fetchUsedSegment(String dataSource, String segmentId)
+  public ListenableFuture<DataSegment> fetchSegment(String dataSource, String segmentId, boolean includeUnused)
   {
     final String path = StringUtils.format(
-        "/druid/coordinator/v1/metadata/datasources/%s/segments/%s",
+        "/druid/coordinator/v1/metadata/datasources/%s/segments/%s%s",
```

Comment:
This looks better to read. We can adjust %s to true/false based on the flag passed. Wdyt?
```suggestion
        "/druid/coordinator/v1/metadata/datasources/%s/segments/%s?includeUnused=%s",
```

Reply:
I added it this way for consistency with the other APIs. The current pattern is to add a flag without value and check if it has been passed or not. ("includeOvershadowedStatus", "full", etc). Do we want to start moving away from this?

Reply:
Something like https://github.com/apache/druid/blob/82a8529402c0237269bb8173fdd58f43935a72c8/server/src/main/java/org/apache/druid/client/coordinator/CoordinatorClientImpl.java#L57 is more readable.

Reply:
I agree with that. Changing to this pattern.